integrations/transportation/trainline.js

```diff
@@ -0,0 +1,149 @@
+    const tripId = trip.booking.id;
+    if (get(trip, 'booking.outward')) {
+      const outwardDate = get(trip, 'booking.outward.date');
+      activities.push({
+        id: `${tripId}-${get(trip, 'booking.outward.origin.id')}`, // a string that uniquely represents this activity
+        datetime: outwardDate, // a javascript Date object that represents the start of the activity
+        durationHours: get(trip, 'booking.outward.duration') / 60, // a floating point that represents the duration of the activity in decimal hours
+        activityType: ACTIVITY_TYPE_TRANSPORTATION,
+        transportationMode: matchTransportMode(get(trip, 'booking.outward.legs[0].transportMode')), // a variable (from definitions.js) that represents the transportation mode
+        departureStation: get(trip, 'booking.outward.origin.name'), // (for other travel types) a string that represents the original starting point
+        destinationStation: get(trip, 'booking.outward.destination.name'), // (for other travel types) a string that represents the final destination
+      });
+    }
+    if (get(trip, 'booking.inward')) {
+      // sometimes inward journeys are open so don't have a fixed return date. Use the outward date instead
+      const inwardDate = get(trip, 'booking.inward.date') || get(trip, 'booking.outward.date');
+
+      // sometimes inward journeys are open so don't have a fixed return date. Use the outward duration instead
+      const inwardDuration = (get(trip, 'booking.inward.duration') || get(trip, 'booking.outward.duration')) / 60;
+
+      // sometimes inward journeys are open so don't have a fixed return date. Use the outward transport type instead
+      const inwardTransportationMode = get(trip, 'booking.inward.legs[0].transportMode') || get(trip, 'booking.outward.legs[0].transportMode');
+
+      activities.push({
+        id: `${tripId}-${get(trip, 'booking.inward.origin.id')}`, // a string that uniquely represents this activity
+        datetime: inwardDate, // a javascript Date object that represents the start of the activity
+        durationHours: inwardDuration, // a floating point that represents the duration of the activity in decimal hours
+        activityType: ACTIVITY_TYPE_TRANSPORTATION,
+        transportationMode: matchTransportMode(inwardTransportationMode), // a variable (from definitions.js) that represents the transportation mode
+        departureStation: get(trip, 'booking.inward.origin.name'), // (for other travel types) a string that represents the original starting point
+        destinationStation: get(trip, 'booking.inward.destination.name'), // (for other travel types) a string that represents the final destination
+      });
+    }
+  });
+  return {
+    activities,
+    newState,
+  };
+}
+
+const config = {
+  label: 'Trainline',
+  country: 'UK',
+  type: ACTIVITY_TYPE_TRANSPORTATION,
+  description: 'collects trips from your train and bus journeys',
+  isPrivate: true,
+  signupLink: null,
```

Comment:
Isn't there a link we could provide here?

Reply:
Yep, have added